### ISSUE TITLE
Breaking change in v0.5.4 (Slack: bot.startRTM())

### ISSUE THREAD

**Issue by gswalden** [howdyai/botkit#829]
The `bot.startRTM()` method was [changed](https://github.com/howdyai/botkit/commit/20e9b619a6b376a41ae82e752c15383a01aebbeb#diff-3d076de69c4172b993d7eaeb5515b3b6L132) to use [rtm.connect](https://api.slack.com/methods/rtm.connect) instead of [rtm.start](https://api.slack.com/methods/rtm.start). However, you can see from those two links the payloads returned differ significantly, notably `rtm.connect` does not contain any user/channel information. The docs still maintain the rtm.start behavior ([link](https://github.com/howdyai/botkit/blob/master/docs/readme-slack.md#botstartrtm)) as well.

This broke one of my bots, which used the user/channel data to populate the storage. As it looks like this change was made to correct an unrelated websocket bug, I'm not sure what the best solution is. Perhaps just updating the docs and making clearer note of this change.

**Comment by jonchurch** [comment]
The other fields available in rtm.start are "officially" unsupported by slack, a slack rep mentioned that many moons ago in the slack-api channel of the dev4slack.slack.com group.

However, there was no indication of this, as far as I can tell, within the documentation here.

**Comment by tmundt** [comment]
To clarify: @gswalden Did you find  a workaround for this? How can it be achieved to connect all teams per connection on RTMstart or something equivalent (IMHO it seems that the shift away from RTM (websocket) to event-based HTTP-POST/GET methods has some disadvantages?)
@jonchurch can we neglect RTM completely in the future for bot connection?

**Comment by jonchurch** [comment]
Currently the Events Api cannot turn off typing indicators. Events api can send indicators, but it is not turned off properly when a message is sent to slack via events api. Slack is aware of this, and the community has been trying to prod them till they fix this. 

Thats the last piece that Events API is missing, in terms of feature parity RTM for with typical bot functions. 

I am unaware of the usecase that @gswalden is implementing here, but honestly I am far more familar with the FB api than the slack one at this point. 

@gawalden, can you elaborate on your usecase here? Can you change to getting that info from an api call to slack to get team info? How badly did this affect you?

**Comment by gswalden** [comment]
The solution was very simple for our use case (our bot is built for a specific team and is not a public app). Prior to calling `bot.startRTM()`, I called `bot.api.users.list()` then saved the results via `controller.storage.users.save()`. So two extra lines of code, nothing serious.

I had spent 15-20 minutes looking into why the initial bug was occurring, so my goal of this issue was more to raise awareness in case others relied on this callback payload.
